pathway-analysis/qusage_replicate_vignette.Rmd

```diff
@@ -0,0 +1,404 @@
+a boxplot of the gene `387` which has a negative value and is on right side of 
+the graph.
+
+```{r}
+graphics::boxplot(exprs_mat["387", ] ~ filtered_metadata_df$timepoint)
+```
+
+### Two way comparison
+
+Patients in this influenza data set were either _symptomatic_ or _asymptomatic_.
+We can check if there are differences in pathway _responses_ to influenza virus
+(e.g., between time points) between the two groups.
+The ability to do this more complex comparison is a great feature of `qusage`.
+Again, we'll mirror the analysis performed in the vignette.
+
+The symptomatic or asymptomatic information is in the `clinic_pheno` column
+of the metadata.
+
+```{r}
+head(filtered_metadata_df$clinic_pheno)
+```
+
+Generate labels for the two way comparison.
+
+```{r}
+twoway_labels <- paste(filtered_metadata_df$clinic_pheno, 
+                       filtered_metadata_df$timepoint,
+                       sep = ".")
+twoway_labels
+```
+
+Symptomatic comparison
+
+```{r}
+sx_results <- qusage(eset = exprs_mat,
+                     labels = twoway_labels,
+                     contrast = "Symptomatic.t1-Symptomatic.t0",
+                     # timepoints are paired samples, so we can specify that
+                     # information here
+                     geneSets = kegg_genesets,
+                     pairVector = filtered_metadata_df$subject)
+```
+
+Asymptomatic comparison
+
+```{r}
+asx_results <- qusage(eset = exprs_mat,
+                      labels = twoway_labels,
+                      contrast = "Asymptomatic.t1-Asymptomatic.t0",
+                      # timepoints are paired samples, so we can specify that
+                      # information here
+                      geneSets = kegg_genesets,
+                      pairVector = filtered_metadata_df$subject)
+```
+
+Calculate the difference between the two comparisons
+
+```{r}
+asx_vs_sx <- qusage(eset = exprs_mat, 
+                    labels = twoway_labels, 
+                    contrast = "(Symptomatic.t1-Symptomatic.t0) - (Asymptomatic.t1-Asymptomatic.t0)", 
+                    geneSets = kegg_genesets, 
+                    pairVector = filtered_metadata_df$subject)
+```
+
+Let's look at the cytosolic DNA sensing pathway like the authors' do in the
+vignette and see if we get similar results.
+
+```{r}
+names(kegg_genesets)[125]
+```
+
+```{r}
+plotDensityCurves(asx_results, 
+                  path.index = 125, 
+                  col = "#D55E00",
+                  main="CYTOSOLIC DNA SENSING",
+                  xlim = c(-1.0, 2.5))
+plotDensityCurves(sx_results, path.index = 125, col = "#56B4E9", add = TRUE)
+plotDensityCurves(asx_vs_sx, path.index = 125, col = "#000000", add = TRUE, 
+                  lwd = 3)
+legend("topleft", 
+       legend =  c("Asymptomatic", "Symptomatic", "Symptomatic-Asymptomatic"),
+       lty = 1, col = c("#D55E00", "#56B4E9", "#000000", lwd = c(1, 1, 3)),
+       cex = 0.5)
+```
```

Comment:
Are we expecting this plot to be *exactly* like the one on page 12 of the vignette? The scale looks a bit different, although the conclusion is and relative curve positions and shapes are mostly the same.

Reply:
Is there a scaling parameter that has been changed as to what the defaults?

Reply:
Never mind. I forgot completely that we have used refine.bio processed data and the authors used whatever pipeline they used for their data in the vignette. This result looks like what we want then.